### PR TITLE
SAS-87 # Acceptance tests - add team manager

### PR DESCRIPTION
**Pull request by Chen2908**

### CODE REVIEW

**Review by AmitNachimovitz — APPROVED** [review]

File: src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java [hunk 1]
```diff
@@ -372,9 +373,76 @@ public void addAdditionalTeamOwnerFail() {
...
+        ((Player)player).setDateOfBirth(LocalDate.parse("1990-12-01"));
+        team.addPlayerToTeam((Player)player);
+
+        Team myTeam = ((TeamOwner) teamOwner).getTeam();
+        myTeam.getPersonalPage().showPersonalPage();
+
+        //enter editing mode
+        if (teamManagement.enterEditingMode(teamOwner, myTeam)) {
+            System.out.println("--Select to add a team manager--");
+            System.out.println("Please select a nominee from the followings:");
```
Comment (yaarm) [inline src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java:393]: Missing check - if the user can add team owners to the team.

File: src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java [hunk 2]
```diff
@@ -372,9 +373,76 @@ public void addAdditionalTeamOwnerFail() {
...
+            if (toMakeTeamOwner !=null){
+                toMakeTeamOwner = teamManagement.addTeamManager(toMakeTeamOwner, myTeam, teamOwner, givePrivileges);
+                assertTrue(toMakeTeamOwner instanceof TeamManager);
+                System.out.println(((TeamManager)toMakeTeamOwner).getFullName() + " was nominated as team manager");
+            }else{
+                System.out.println("The team manager could not be nominated");
+            }
+        }
+        else {
+            System.out.println("You do not have the privileges to add the team manager");
```
Comment (yaarm) [inline src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java]: This should be - you don't have editor permissions.

File: src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java [hunk 3]
```diff
@@ -372,9 +373,76 @@ public void addAdditionalTeamOwnerFail() {
...
+        ((Player)player).setTeam(team);
+        ((Player)player).setDateOfBirth(LocalDate.parse("1990-12-01"));
+        team.addPlayerToTeam((Player)player);
+
+        Team myTeam = ((TeamOwner) teamOwner).getTeam();
+        myTeam.getPersonalPage().showPersonalPage();
+
+        //enter editing mode
+        if (teamManagement.enterEditingMode(teamOwner, myTeam)) {
+            System.out.println("--Select to add a team manager--");
```
Comment (yaarm) [inline src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java:428]: Same here - just add the check if he is authorize to add team manager.

File: src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java [hunk 4]
```diff
@@ -372,9 +373,76 @@ public void addAdditionalTeamOwnerFail() {
...
+            String approval = "1";
+            System.out.println(approval);
+            boolean givePrivileges = approval.equals("1") ? true : false;
+            User toMakeTeamOwner = teamManagement.getUserForTeamManagerNominees(myTeam, "");
+            if (toMakeTeamOwner !=null){
+                toMakeTeamOwner = teamManagement.addTeamManager(toMakeTeamOwner, myTeam, teamOwner, givePrivileges);
+                assertTrue(toMakeTeamOwner instanceof TeamManager);
+                System.out.println(((TeamManager)toMakeTeamOwner).getFullName() + " was nominated as team manager");
+            }else{
+                System.out.println("The team manager could not be nominated");
```
Comment (yaarm) [inline src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java:442]: Missing assert of the team manager wasn't nominated - as:
assertNull(team.getTeamManager());

Comment (Chen2908) [reply src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java:442]: Will fix it

File: src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java (hunk 1 above)
Comment (Chen2908) [reply src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java:393]: Why? It is checked inside the function.. Why check twice?

Comment (yaarm) [reply src/test/java/com/SAS/acceptanceTests/TeamManagementAT.java:393]: Ok.

**Review by yaarm — APPROVED** [review]

**Pull request merged**